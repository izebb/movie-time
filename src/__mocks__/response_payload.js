export const movieListPayload = [
  [
    {
      images: {
        hero: {
          template:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/279/80/GreysAnatomy14_Packshot.jpg{?width,height}&template=abcstudios',
          url:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/279/80/GreysAnatomy14_Packshot.jpg?width=960&height=540&template=abcstudios'
        },
        lanscape: {
          template:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/279/80/GreysAnatomy14_Packshot.jpg{?width,height}&template=abcstudios',
          url:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/279/80/GreysAnatomy14_Packshot.jpg?width=960&height=540&template=abcstudios'
        }
      },
      rating: {
        id: 'tt0413573',
        rating: '7.6',
        url: 'http://www.imdb.com/title/tt0413573?ref_ext_viaplay',
        votes: '207 500'
      },
      series: {
        seasons: 14,
        seriesGuid: 'series-greys-anatomy',
        synopsis:
          'Ett högintensivt medicinskt drama där vi får följa Meredith Grey och hennes kollegor. De ställs dagligen inför livsavgörande beslut, men hos varandra hittar de stöd, vänskap och en hel del kärlek.',
        title: "Grey's Anatomy"
      }
    },
    {
      images: {
        hero: {
          template:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/712/784/CriminalMinds_Landscape.jpg{?width,height}&template=abcstudios',
          url:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/712/784/CriminalMinds_Landscape.jpg?width=960&height=540&template=abcstudios'
        },
        lanscape: {
          template:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/712/784/CriminalMinds_Landscape.jpg{?width,height}&template=abcstudios',
          url:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/712/784/CriminalMinds_Landscape.jpg?width=960&height=540&template=abcstudios'
        }
      },
      rating: {
        id: 'tt0452046',
        rating: '8.1',
        url: 'http://www.imdb.com/title/tt0452046?ref_ext_viaplay',
        votes: '136 070'
      },
      series: {
        seasons: 13,
        seriesGuid: 'series-criminal-minds',
        synopsis:
          'FBI:s elitteam, the Behavioral Analysis Unit, utreder USA:s farligaste kriminella hjärnor och försöker förutse gärningsmännens nästa drag – och stoppa dem innan de hinner slå till igen.',
        title: 'Criminal Minds'
      }
    },
    {
      images: {
        hero: {
          template:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/599/188/TheBlacklist6Exc_Packshot.jpg{?width,height}',
          url:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/599/188/TheBlacklist6Exc_Packshot.jpg?width=960&height=540'
        },
        lanscape: {
          template:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/599/188/TheBlacklist6Exc_Packshot.jpg{?width,height}',
          url:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/599/188/TheBlacklist6Exc_Packshot.jpg?width=960&height=540'
        }
      },
      rating: {
        id: 'tt2741602',
        rating: '8.1',
        url: 'http://www.imdb.com/title/tt2741602?ref_ext_viaplay',
        votes: '152 741'
      },
      series: {
        availabilityInfo: '9 Mars 05.00',
        seasons: 6,
        seriesGuid: 'series-blacklist-the',
        synopsis:
          'I en av Viaplays mest populära serier följer vi FBI-agenten Elizabeth Keen, som under mystiska omständigheter möter det kriminella geniet Red. Tillsammans bekämpar de världens farligaste brottslingar.',
        title: 'The Blacklist'
      }
    },
    {
      images: {
        hero: {
          template:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/444/784/TheVampireDiaries8_Packshot.jpg{?width,height}',
          url:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/444/784/TheVampireDiaries8_Packshot.jpg?width=960&height=540'
        },
        lanscape: {
          template:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/444/784/TheVampireDiaries8_Packshot.jpg{?width,height}',
          url:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/444/784/TheVampireDiaries8_Packshot.jpg?width=960&height=540'
        }
      },
      rating: {
        id: 'tt1405406',
        rating: '7.7',
        url: 'http://www.imdb.com/title/tt1405406?ref_ext_viaplay',
        votes: '259 135'
      },
      series: {
        seasons: 8,
        seriesGuid: 'series-vampire-diaries-the',
        synopsis:
          'Två vampyrbröder, en ond och en god, återvänder till hemstaden Mystic Falls. Där förälskar de sig i samma tjej, Elena. Kampen är ett faktum och nu är det upp till Elena – vem ska hon välja?',
        title: 'The Vampire Diaries'
      }
    }
  ],
  [
    {
      images: {
        hero: {
          template:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/286/700/LethalWeapon3Exc_Packshot.jpg{?width,height}',
          url:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/286/700/LethalWeapon3Exc_Packshot.jpg?width=960&height=540'
        },
        lanscape: {
          template:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/286/700/LethalWeapon3Exc_Packshot.jpg{?width,height}',
          url:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/286/700/LethalWeapon3Exc_Packshot.jpg?width=960&height=540'
        }
      },
      rating: {
        id: 'tt5164196',
        rating: '8.0',
        url: 'http://www.imdb.com/title/tt5164196?ref_ext_viaplay',
        votes: '31 479'
      },
      series: {
        seasons: 3,
        seriesGuid: 'series-lethal-weapon',
        synopsis:
          'En av Hollywoods mest efterlängtade nyinspelningar och baserad på filmerna med samma namn. En kaxig polisduo jobbar hårt för att hålla ställningarna i ett pulserande L.A.',
        title: 'Lethal Weapon'
      }
    },
    {
      images: {
        hero: {
          template:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/118/792/chicagofire5_packshot.jpg{?width,height}',
          url:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/118/792/chicagofire5_packshot.jpg?width=960&height=540'
        },
        lanscape: {
          template:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/118/792/chicagofire5_packshot.jpg{?width,height}',
          url:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/118/792/chicagofire5_packshot.jpg?width=960&height=540'
        }
      },
      rating: {
        id: 'tt2261391',
        rating: '7.9',
        url: 'http://www.imdb.com/title/tt2261391?ref_ext_viaplay',
        votes: '29 842'
      },
      series: {
        seasons: 5,
        seriesGuid: 'series-chicago-fire',
        synopsis:
          'För de modiga brandmännen på Brandstation 51 kommer alltid offrens säkerhet först. Följ dem i deras dagliga kamp fylld av dramatik, spänning och interna stridigheter.',
        title: 'Chicago Fire'
      }
    },
    {
      images: {
        hero: {
          template:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/935/376/FamilyGuy_landscape_GE.jpg{?width,height}',
          url:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/935/376/FamilyGuy_landscape_GE.jpg?width=960&height=540'
        },
        lanscape: {
          template:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/935/376/FamilyGuy_landscape_GE.jpg{?width,height}',
          url:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/935/376/FamilyGuy_landscape_GE.jpg?width=960&height=540'
        }
      },
      rating: {
        id: 'tt0182576',
        rating: '8.2',
        url: 'http://www.imdb.com/title/tt0182576?ref_ext_viaplay',
        votes: '279 710'
      },
      series: {
        seasons: 16,
        seriesGuid: 'series-family-guy',
        synopsis:
          'Denna kultserie, med fler än hundra nomineringar och priser, är en av världens mest älskade. Peter Griffin och hans dysfunktionella familj bjuder på knäppa vardagsscenarios som du sent kommer glömma.',
        title: 'Family Guy'
      }
    },
    {
      images: {
        hero: {
          template:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/605/816/Blindspot4Exc_Packshot.jpg{?width,height}',
          url:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/605/816/Blindspot4Exc_Packshot.jpg?width=960&height=540'
        },
        lanscape: {
          template:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/605/816/Blindspot4Exc_Packshot.jpg{?width,height}',
          url:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/605/816/Blindspot4Exc_Packshot.jpg?width=960&height=540'
        }
      },
      rating: {
        id: 'tt4474344',
        rating: '7.5',
        url: 'http://www.imdb.com/title/tt4474344?ref_ext_viaplay',
        votes: '58 938'
      },
      series: {
        availabilityInfo: '11 Mars 20.00',
        seasons: 4,
        seriesGuid: 'series-blindspot',
        synopsis:
          'Ett konspiratoriskt kriminaldrama som kretsar kring en kvinna som hittas på Times Square, utan minne och med kroppen täckt av mystiska tatueringar.',
        title: 'Blindspot'
      }
    }
  ],
  [
    {
      images: {
        hero: {
          template:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/407/556/TheGoodDoctor2Exc_packshot.jpg{?width,height}',
          url:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/407/556/TheGoodDoctor2Exc_packshot.jpg?width=960&height=540'
        },
        lanscape: {
          template:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/407/556/TheGoodDoctor2Exc_packshot.jpg{?width,height}',
          url:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/407/556/TheGoodDoctor2Exc_packshot.jpg?width=960&height=540'
        }
      },
      rating: {
        id: 'tt6470478',
        rating: '8.3',
        url: 'http://www.imdb.com/title/tt6470478?ref_ext_viaplay',
        votes: '41 578'
      },
      series: {
        availabilityInfo: 'Tisdagar 20.00',
        seasons: 2,
        seriesGuid: 'series-good-doctor-the',
        synopsis:
          'Kan en person som saknar förmågan att relatera till människor faktiskt rädda liv? En ung kirurg med svårigheter att kommunicera läker oron hos sina patienter genom sin smått övernaturliga kunskap.',
        title: 'The Good Doctor'
      }
    },
    {
      images: {
        hero: {
          template:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/610/844/SWAT2Exc_Packshot.jpg{?width,height}',
          url:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/610/844/SWAT2Exc_Packshot.jpg?width=960&height=540'
        },
        lanscape: {
          template:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/610/844/SWAT2Exc_Packshot.jpg{?width,height}',
          url:
            'https://i-viaplay-com.akamaized.net/viaplay-prod/610/844/SWAT2Exc_Packshot.jpg?width=960&height=540'
        }
      },
      rating: {
        id: 'tt6111130',
        rating: '6.7',
        url: 'http://www.imdb.com/title/tt6111130?ref_ext_viaplay',
        votes: '8 415'
      },
      series: {
        availabilityInfo: '8 Mars 20.00',
        seasons: 2,
        seriesGuid: 'series-s.w.a.t.',
        synopsis:
          'En ung mörkhyad man skjuts av misstag av ledaren för S.W.A.T och Daniel ”Hondo” Harrelsson, som är född och uppvuxen i området, tvingas att axla rollen som ny ledare. Hondo slits mellan lojaliteten till gatan och plikten till sina medarbetare i strävan efter att ena samhället som är i uppror.',
        title: 'S.W.A.T.'
      }
    }
  ]
];
