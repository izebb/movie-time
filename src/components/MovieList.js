import React from 'react';
import styled from 'styled-components';

import { MovieCard } from './MovieCard';
import { MovieSummary } from './MovieSummary';

const ListSection = styled.div`
  display: flex;
  flex-wrap: wrap;
  padding: 0 20px;
`;

const CardWrapper = styled.div`
  padding: 10px;
`;

const Container = styled.div`
  max-width: 1200px;
  margin: 0 auto;
`;

export const MovieList = ({
  list,
  onClickMovie,
  clearMovieSelection,
  displaySection,
  displaSectionIndex
}) => {
  return (
    <Container>
      {list.map((movies, index) => {
        return (
          <div key={index}>
            <ListSection>
              {movies.map(movie => {
                return (
                  <CardWrapper key={movie.series.seriesGuid}>
                    <MovieCard
                      id={movie.series.seriesGuid}
                      onClick={onClickMovie}
                      title={movie.series.title}
                      seasons={movie.series.seasons}
                      coverTemplate={movie.images.lanscape.template}
                    />
                  </CardWrapper>
                );
              })}
            </ListSection>
            <MovieSummary
              onClear={clearMovieSelection}
              show={displaSectionIndex === index}
              summary={displaySection.series}
              rating={displaySection.rating}
              clearMovieSelection={clearMovieSelection}
              images={displaySection.images}
            />
          </div>
        );
      })}
    </Container>
  );
};

MovieList.defaultProps = {
  list: [],
  onClickMovie: () => {},
  clearMovieSelection: () => {},
  displaySection: {},
  displaSectionIndex: -1
};
