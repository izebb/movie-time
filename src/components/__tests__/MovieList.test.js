import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { MovieList } from '../MovieList';
import { movieListPayload } from '../../__mocks__/response_payload';

describe('MovieList', () => {
  it('should render without crashing', () => {
    const props = {
      list: movieListPayload
    };
    const wrapper = shallow(<MovieList {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
