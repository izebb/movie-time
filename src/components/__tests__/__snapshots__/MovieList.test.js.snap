// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList should render without crashing 1`] = `
<MovieList__Container>
  <div
    key="0"
  >
    <MovieList__ListSection>
      <MovieList__CardWrapper
        key="series-greys-anatomy"
      >
        <MovieCard
          coverTemplate="https://i-viaplay-com.akamaized.net/viaplay-prod/279/80/GreysAnatomy14_Packshot.jpg{?width,height}&template=abcstudios"
          id="series-greys-anatomy"
          onClick={[Function]}
          seasons={14}
          title="Grey's Anatomy"
        />
      </MovieList__CardWrapper>
      <MovieList__CardWrapper
        key="series-criminal-minds"
      >
        <MovieCard
          coverTemplate="https://i-viaplay-com.akamaized.net/viaplay-prod/712/784/CriminalMinds_Landscape.jpg{?width,height}&template=abcstudios"
          id="series-criminal-minds"
          onClick={[Function]}
          seasons={13}
          title="Criminal Minds"
        />
      </MovieList__CardWrapper>
      <MovieList__CardWrapper
        key="series-blacklist-the"
      >
        <MovieCard
          coverTemplate="https://i-viaplay-com.akamaized.net/viaplay-prod/599/188/TheBlacklist6Exc_Packshot.jpg{?width,height}"
          id="series-blacklist-the"
          onClick={[Function]}
          seasons={6}
          title="The Blacklist"
        />
      </MovieList__CardWrapper>
      <MovieList__CardWrapper
        key="series-vampire-diaries-the"
      >
        <MovieCard
          coverTemplate="https://i-viaplay-com.akamaized.net/viaplay-prod/444/784/TheVampireDiaries8_Packshot.jpg{?width,height}"
          id="series-vampire-diaries-the"
          onClick={[Function]}
          seasons={8}
          title="The Vampire Diaries"
        />
      </MovieList__CardWrapper>
    </MovieList__ListSection>
    <MovieSummary
      clearMovieSelection={[Function]}
      images={Object {}}
      onClear={[Function]}
      rating={Object {}}
      show={false}
      summary={Object {}}
    />
  </div>
  <div
    key="1"
  >
    <MovieList__ListSection>
      <MovieList__CardWrapper
        key="series-lethal-weapon"
      >
        <MovieCard
          coverTemplate="https://i-viaplay-com.akamaized.net/viaplay-prod/286/700/LethalWeapon3Exc_Packshot.jpg{?width,height}"
          id="series-lethal-weapon"
          onClick={[Function]}
          seasons={3}
          title="Lethal Weapon"
        />
      </MovieList__CardWrapper>
      <MovieList__CardWrapper
        key="series-chicago-fire"
      >
        <MovieCard
          coverTemplate="https://i-viaplay-com.akamaized.net/viaplay-prod/118/792/chicagofire5_packshot.jpg{?width,height}"
          id="series-chicago-fire"
          onClick={[Function]}
          seasons={5}
          title="Chicago Fire"
        />
      </MovieList__CardWrapper>
      <MovieList__CardWrapper
        key="series-family-guy"
      >
        <MovieCard
          coverTemplate="https://i-viaplay-com.akamaized.net/viaplay-prod/935/376/FamilyGuy_landscape_GE.jpg{?width,height}"
          id="series-family-guy"
          onClick={[Function]}
          seasons={16}
          title="Family Guy"
        />
      </MovieList__CardWrapper>
      <MovieList__CardWrapper
        key="series-blindspot"
      >
        <MovieCard
          coverTemplate="https://i-viaplay-com.akamaized.net/viaplay-prod/605/816/Blindspot4Exc_Packshot.jpg{?width,height}"
          id="series-blindspot"
          onClick={[Function]}
          seasons={4}
          title="Blindspot"
        />
      </MovieList__CardWrapper>
    </MovieList__ListSection>
    <MovieSummary
      clearMovieSelection={[Function]}
      images={Object {}}
      onClear={[Function]}
      rating={Object {}}
      show={false}
      summary={Object {}}
    />
  </div>
  <div
    key="2"
  >
    <MovieList__ListSection>
      <MovieList__CardWrapper
        key="series-good-doctor-the"
      >
        <MovieCard
          coverTemplate="https://i-viaplay-com.akamaized.net/viaplay-prod/407/556/TheGoodDoctor2Exc_packshot.jpg{?width,height}"
          id="series-good-doctor-the"
          onClick={[Function]}
          seasons={2}
          title="The Good Doctor"
        />
      </MovieList__CardWrapper>
      <MovieList__CardWrapper
        key="series-s.w.a.t."
      >
        <MovieCard
          coverTemplate="https://i-viaplay-com.akamaized.net/viaplay-prod/610/844/SWAT2Exc_Packshot.jpg{?width,height}"
          id="series-s.w.a.t."
          onClick={[Function]}
          seasons={2}
          title="S.W.A.T."
        />
      </MovieList__CardWrapper>
    </MovieList__ListSection>
    <MovieSummary
      clearMovieSelection={[Function]}
      images={Object {}}
      onClear={[Function]}
      rating={Object {}}
      show={false}
      summary={Object {}}
    />
  </div>
</MovieList__Container>
`;
